src/lfx/src/lfx/utils/ssrf_protection.py

```diff
@@ -0,0 +1,324 @@
+- Loopback addresses
+- Cloud metadata endpoints (169.254.169.254)
+- Other internal/special-use addresses
+
+IMPORTANT: HTTP Redirects
+    According to OWASP SSRF Prevention Cheat Sheet, HTTP redirects should be DISABLED
+    to prevent bypass attacks where a public URL redirects to internal resources.
+    The API Request component has (as of v1.7.0) follow_redirects=False by default.
+    See: https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html
+
+Configuration:
+    LANGFLOW_SSRF_PROTECTION_ENABLED: Enable/disable SSRF protection (default: false)
+        TODO: Change default to true in next major version (2.0)
+    LANGFLOW_SSRF_ALLOWED_HOSTS: Comma-separated list of allowed hosts/CIDR ranges
+        Examples: "192.168.1.0/24,internal-api.company.local,10.0.0.5"
+
+TODO: In next major version (2.0):
+    - Change LANGFLOW_SSRF_PROTECTION_ENABLED default to "true"
+    - Remove warning-only mode and enforce blocking
+    - Update documentation to reflect breaking change
+"""
+
+import ipaddress
+import socket
+from urllib.parse import urlparse
+
+from lfx.logging import logger
+from lfx.services.deps import get_settings_service
+
+
+class SSRFProtectionError(ValueError):
+    """Raised when a URL is blocked due to SSRF protection."""
+
+
+# Define blocked IP ranges
+BLOCKED_IP_RANGES = [
+    # IPv4 ranges
+    ipaddress.ip_network("0.0.0.0/8"),  # Current network (only valid as source)
+    ipaddress.ip_network("10.0.0.0/8"),  # Private network (RFC 1918)
+    ipaddress.ip_network("100.64.0.0/10"),  # Carrier-grade NAT (RFC 6598)
+    ipaddress.ip_network("127.0.0.0/8"),  # Loopback
+    ipaddress.ip_network("169.254.0.0/16"),  # Link-local / AWS metadata
+    ipaddress.ip_network("172.16.0.0/12"),  # Private network (RFC 1918)
+    ipaddress.ip_network("192.0.0.0/24"),  # IETF Protocol Assignments
+    ipaddress.ip_network("192.0.2.0/24"),  # Documentation (TEST-NET-1)
+    ipaddress.ip_network("192.168.0.0/16"),  # Private network (RFC 1918)
+    ipaddress.ip_network("198.18.0.0/15"),  # Benchmarking
+    ipaddress.ip_network("198.51.100.0/24"),  # Documentation (TEST-NET-2)
+    ipaddress.ip_network("203.0.113.0/24"),  # Documentation (TEST-NET-3)
+    ipaddress.ip_network("224.0.0.0/4"),  # Multicast
+    ipaddress.ip_network("240.0.0.0/4"),  # Reserved
+    ipaddress.ip_network("255.255.255.255/32"),  # Broadcast
+    # IPv6 ranges
+    ipaddress.ip_network("::1/128"),  # Loopback
+    ipaddress.ip_network("::/128"),  # Unspecified address
+    ipaddress.ip_network("::ffff:0:0/96"),  # IPv4-mapped IPv6 addresses
+    ipaddress.ip_network("100::/64"),  # Discard prefix
+    ipaddress.ip_network("2001::/23"),  # IETF Protocol Assignments
+    ipaddress.ip_network("2001:db8::/32"),  # Documentation
+    ipaddress.ip_network("fc00::/7"),  # Unique local addresses (ULA)
+    ipaddress.ip_network("fe80::/10"),  # Link-local
+    ipaddress.ip_network("ff00::/8"),  # Multicast
+]
```

Comment:
Maybe set this in a function using a global var? just to avoid impacting startup time.